lib/datadog/core/telemetry/event.rb

```diff
@@ -0,0 +1,68 @@
+# typed: true
+
+require 'datadog/core/telemetry/collector'
+require 'datadog/core/telemetry/v1/app_started'
+require 'datadog/core/telemetry/v1/telemetry_request'
+
+module Datadog
+  module Core
+    module Telemetry
+      # Class defining methods to construct a Telemetry event
+      class Event
+        include Kernel
+
+        include Telemetry::Collector
+
+        API_VERSION = 'v1'.freeze
+        ALLOWED_API_VERSIONS = ['v1'.freeze].freeze
+        ERROR_BAD_API_VERSION = ":api_version must be one of ['v1']".freeze
+
+        attr_reader \
+          :api_version
+
+        # @param api_version [String] telemetry API version to request; defaults to `v1`
+        def initialize(api_version: API_VERSION)
+          raise ArgumentError, ERROR_BAD_API_VERSION unless ALLOWED_API_VERSIONS.include?(api_version)
+
+          @api_version = api_version
+        end
+
+        # Forms a TelemetryRequest object based on the event request_type
+        # @param request_type [String] the type of telemetry request to collect data for
+        # @param seq_id [Integer] the ID of the request; incremented each time a telemetry request is sent to the API
+        def telemetry_request(request_type:, seq_id:)
+          Telemetry::V1::TelemetryRequest.new(
+            api_version: @api_version,
```

Comment:
`Telemetry::V1` tells me this is only for the V1 version.
But `api_version: @api_version` tells me the version is configurable.

Does it make sense to hard-code `@api_version = 'v1'` inside `Telemetry::V1::TelemetryRequest.new`?